src/Route.svelte

```diff
@@ -66,6 +68,24 @@
     activeProps = getProps($$props, thisProps);
   }
 
+  $: if (activeRouter) {
+    if (!component) { // component passed as slot
+      hasLoaded = true;
+    } else if (isSvelteComponent(component)) { // component passed as Svelte component
+      hasLoaded = true;
+    } else if (isPromise(component)) { // component passed as import()
+      component.then(module => {
+        component = module.default;
+        hasLoaded = true;
+      });
+    } else if (isFunction(component) && activeRouter.path === path) { // component passed as () => import()
```

Comment:
Is not the `isFunction()` check too redundant at this point?

Reply:
Yes, this one is an unnecessary extra safety :). Removed.